libpromises/match_scope.h

```diff
@@ -26,6 +26,10 @@
 #define CFENGINE_MATCH_SCOPE_H
 
 #include <cf3.defs.h>
+#include <regex.h>
```

Comment:
Yeah, this file looks like it provides generic regex facilities, so go ahead and move it. Even to libutils if possible, but even libpromises sounds better than cf-agent.